### ISSUE TITLE
Question: is there any method to check whether a client is connected or not

### ISSUE THREAD

**Issue by bodycombathuang** [brianc/node-postgres#3010]
const client=new Client("connectstring";
I know client has connect and end methods to connect to or disconnect from db

1.is there any method to check whether client is in connected state or disconnected state?
2. if client.end() is called,  client.connect() cannot be called ,right? I have to create a new instance of Client to connect to db?

**Comment by ManchukondaManoj** [comment]
Yes, we have to re-intialize the connection. If you have proper connection string and your db is connected, Better use connection pool and always keep 1 or 2 connections open.

**Comment by Amadeea** [comment]
1. use client.connection._ending which have boolean value. value will be true if the connection already ended, value will be false if connection not yet ended.
2. yes you must create another client.

CMIIW
